### PR TITLE
Store isGrid option in localStorage

### PR DESCRIPTION
**Pull request by MahmoudHamdy02**
Since we've recently added `showHiddenItems` to localStorage, it also feels logical to include the view mode.

I've added a small test case but I'm not sure why it fails yet.

### CODE REVIEW

File: src/header.jsx [hunk 1]
```diff
@@ -94,7 +94,10 @@ const ViewSelector = ({ isGrid, setIsGrid, sortBy, setSortBy }) => {
                               ? _("Display as a list")
                               : _("Display as a grid")}
                           key="view-toggle-action"
-                          onClick={() => setIsGrid(!isGrid)}
+                          onClick={() => setIsGrid(isGrid => {
+                              localStorage.setItem("cockpit-navigator.isGrid", !isGrid ? "true" : "false");
+                              return !isGrid;
```
Comment (cockpituous) [inline src/header.jsx:99]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-243-20240205-175524-9b6c5678-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by jelly** [comment]
@subhoghoshX do you need help to get this across the finish line, is there anything I can do to help?

**Comment by subhoghoshX** [comment]
> @subhoghoshX do you need help to get this across the finish line, is there anything I can do to help?

I think you meant to tag @MahmoudHamdy02?

**Comment by subhoghoshX** [comment]
> @subhoghoshX do you need help to get this across the finish line, is there anything I can do to help?

I think you meant to tag @MahmoudHamdy02 :) ?

**Comment by jelly** [comment]
> > @subhoghoshX do you need help to get this across the finish line, is there anything I can do to help?
> 
> I think you meant to tag @MahmoudHamdy02 :) ?

Oh woops! Sorry yes :)

**Comment by MahmoudHamdy02** [comment]
Hello! Sorry for leaving this, I'm a bit busy with university :/.

IIRC the local storage code was working fine, but I couldn't figure out why the test case was failing. It would indeed be great if you could take a look :)

File: src/header.jsx (hunk 1 above)
Comment (cockpituous) [inline src/header.jsx:99]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-243-20240306-124050-3aacf570-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]
Thanks!

**Pull request merged**